index.html

```diff
@@ -1100,6 +1070,12 @@
   </p>
   </section> <!-- NFCRecordType interface -->
 
+  <section><h3>The <strong>NFCRecordData</strong> type</h3>
+    <pre class="idl">
+      typedef (DOMString or unrestricted double or object or ArrayBuffer) NFCRecordData;
```

Comment:
Yes. Please put `typedef NFCRecordData` above `dictionary NFCRecord` within the same IDL block as suggested. It will improve readability and is aligned with the platform conventions.

The same with `enum NFCRecordType`, move it above the dictionary within the same IDL block where it is used.

For consistency and to conform with the conventions, also move `callback MessageCallback` *below* `interface NFCAdapter` within the same IDL block. I'd also move `typedef NFCMessage` above that interface too.

Also merge `enum NFCPushTarget` and `dictionary NFCPushOptions`.

Consider merging `enum NFCWatchMode` and `dictionary NFCWatchOptions` if you think that'd improve readability (i.e. consider how the IDL and prose work together vs. split separate).

(Merging definitions was hard or impossible with ReSpec before Contiguous IDL support, thus many specs keep these separate. That is due to a tool limitation.)